services/venice-controller/src/main/java/com/linkedin/venice/controller/server/AbstractRoute.java

```diff
@@ -150,6 +150,23 @@ protected boolean isAllowListUser(Request request) {
     return accessController.get().isAllowlistUsers(certificate, storeName, HTTP_GET);
   }
 
+  /**
+   * Check whether the user is within the admin users allowlist for store deletion.
+   */
+  protected boolean isAllowListUserForStoreDeletion(Request request) {
+    if (!isAclEnabled()) {
+      /**
+       * Grant access if it's not required to check ACL.
+       * {@link accessController} will be empty if ACL is not enabled.
+       */
+      return true;
```

Comment:
We want to do the opposite thing, right? If ACL is not enabled, do not allow the store deletion at all.

Besides, it's cleaner to check whether ACL is enabled in AdminSparkServer - do not register this API if ACL is not enabled:
`
    if (sslEnabled) {
      httpService.post(DELETE_STORE.getPath(),
          new VeniceParentControllerRegionStateHandler(admin, storesRoutes.deleteStore(admin)));
    }
`
The spike will be fixing the test cases; let's see how difficult that is. WDYT?